### PR TITLE
Fix find_library so that it looks in sys.prefix/lib first

### PR DESCRIPTION
**Pull request by tacaswell**
This backports https://github.com/AnacondaRecipes/python-feedstock/commit/68802d1962508a3c731be77abd123fb2ce9c2f54

closes https://github.com/conda-forge/python-feedstock/issues/157

Conflicts:
	recipe/meta.yaml
           The meta.yaml files have significantly
           diverged (AnacondaRecipes includes many more patches +
	   differently handles extra down loads) + bulid number
           differences

### CODE REVIEW

**Comment by jakirkham** [comment]
Thanks @tacaswell. Could you please do the same for the Python 2.7 and 3.5 branches?

**Comment by tacaswell** [comment]
sure

**Comment by tacaswell** [comment]
3.4 and 2.6 are dead?

**Comment by jakirkham** [comment]
If you want to try for those, you are more than welcome to. However we don't actively support 3.4 in conda-forge any more. 2.6 was never supported in conda-forge, but was added for historical reasons.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -22,9 +22,9 @@ source:
     - win-msc_ver-1900.patch
     - fix-ABI-issue29943.patch
     - do-not-download-externals.patch  # [win]
-
+    - 0015-Fix-find_library-so-that-it-looks-in-sys.prefix-lib-.patch
 build:
-  number: 1
+  number: 3
```
Comment (ocefpaf) [inline recipe/meta.yaml]: @tacaswell I merged #158 so we need this to be 3 to get a new package.

Thanks for the PR BTW!

**Comment by jakirkham** [comment]
Thanks @tacaswell for doing all of these. Have just now merged them. Should be built soon.

**Comment by tacaswell** [comment]
@jakirkham no problem!  Hopefully makes up for making no progress on your h5py issue.

**Comment by jakirkham** [comment]
No worries. That was a tricky problem.

This should be very useful to a lot of people, who are otherwise patching packages in `conda-forge` to have a similar effect.

**Pull request merged**